build_tools/ci/cpu_comparison/input_generator.py

```diff
@@ -0,0 +1,151 @@
+#
+# This script also create a file containing a single line of the form
+# `--input="3x40xf32=@<binary_file>" --input="2x2xi32=@<binary_file>"`
+#
+# which will be used as input to iree-run-module in the main script.
+
+
+import numpy as np
+import struct
+import sys
+import os
+
+
+def float32_to_bfloat16(float32_value):
+
+    # Convert to a 32-bit integer representation
+    int32_repr = float32_value.view(np.int32)
+
+    # Truncate the lower 16 bits
+    bf16_int_repr = (int32_repr >> 16) & 0xFFFF
+
+    # Convert to numpy uint16 for consistency
+    return np.uint16(bf16_int_repr)
+
+
+def generate_bfloat16_data(num_values, lower_bound, upper_bound):
+
+    float_data = np.random.randint(lower_bound, upper_bound, num_values).astype(
+        np.float32
+    )
+
+    # Convert float32 data to bfloat16
+    bf16_data = [float32_to_bfloat16(f) for f in float_data]
+
+    # Pack bfloat16 data into binary format
+    binary_data = struct.pack(f"{len(bf16_data)}H", *bf16_data)
+
+    return binary_data
+
+
+def get_numpy_type(element_type):
+    if element_type == "float32" or element_type == "f32":
+        return np.float32
+    elif element_type == "int32" or element_type == "i32":
+        return np.int32
+    elif element_type == "int16" or element_type == "i16":
+        return np.int16
+    elif element_type == "int8" or element_type == "i8":
+        return np.int8
+    elif element_type == "bfloat16" or element_type == "bf16":
+        raise ValueError(
+            "Type 'bfloat16' is not supported along this path through the program (something went wrong)."
+        )
+    else:
+        raise ValueError("Invalid or unsupported element type: " + element_type)
+
+
+def write_input(bin_filename, num_elements, element_type, input_number):
+
+    # Random integer values in range [lower_bound, upper_bound)
+    # will be generated for the input data.
+    lower_bound = 0
+    upper_bound = 5
+
+    # Fix the seed for each input, based on the input number.
+    np.random.seed(1 + input_number)
+
+    data = None
+    if element_type == "bfloat16" or element_type == "bf16":
+        data = generate_bfloat16_data(num_elements, lower_bound, upper_bound)
+    else:
+        dtype = get_numpy_type(element_type)
+        tensor = np.random.randint(lower_bound, upper_bound, num_elements).astype(dtype)
+        # Binary date from 'tensor'
+        data = tensor.tobytes()
+
+    with open(bin_filename, "wb") as file:
+        file.write(data)
+
+
+def generate_inputs(filename, write_dir):
+    """
+    Parse the input file 'filename' and generate binary files for the inputs of
+    the mlir function.
+    """
+
+    name = os.path.splitext(os.path.basename(filename))[0]
+
+    input_args = []
+
+    with open(filename, "r") as file:
+        input_number = 1
+
+        # Looking for lines of the form '# input 3x40xf32'
+        for line in file:
+            tokens = line.split()
+            if len(tokens) > 2 and tokens[0] == "//":
```

Comment:
You should probably add some warning/error message if this condition doesn't hold.